processing/src/main/java/org/apache/druid/query/aggregation/first/NumericFirstVectorAggregator.java

```diff
@@ -132,14 +183,14 @@ public void aggregate(
    *
    * @param buf         byte buffer storing the byte array representation of the aggregate
    * @param position    offset within the byte buffer at which the current aggregate value is stored
-   * @param time        the time to be updated in the buffer as the last time
-   * @param index       the index of the vectorized vector which is the last value
+   * @param time        the time to be updated in the buffer as the first time
+   * @param number      number which is the first value
    */
-  void updateTimeWithValue(ByteBuffer buf, int position, long time, int index)
+  void updateTimeWithValue(ByteBuffer buf, int position, long time, Number number)
```

Comment:
Autoboxing alert, as mentioned in the other comment. Let's see if we can get away without this change.

Reply:
It might not be as clean as the current code, however, there shouldn't be a lot of duplication.